src/python/pants/backend/jvm/tasks/junit_run.py

```diff
@@ -350,10 +350,12 @@ def parse_error_handler(parse_error):
     if result == 0:
       return TestResult.successful
 
+    # NB: If the TestRegistry fails to find the owning target of a failed test, the target key in
+    # this dictionary will be None: helper methods in this block account for that.
     target_to_failed_test = parse_failed_targets(test_registry, output_dir, parse_error_handler)
 
     def sort_owning_target(t):
-      return t.address.spec if t else None
+      return t.address.spec if t else ''
```

Comment:
Even with the NB above, I'd still add 

```
NB: we use the empty string rather than None to ensure comparisons do not fail.
```

Reply:
There are now two comments in this method body, heh. I added one before seeing the one below.

Reply:
Feel free to update the below comment to simply say `Python does not allow` instead of `Python 3` :) 40 more days until Python 2 is officially EOL!